CHANGELOG.md

```diff
@@ -1,5 +1,37 @@
 # Changelog
 
+## [1.4](https://github.com/arborx/arborx/tree/v1.4) (2023-05-05)
+
+[Full Changelog](https://github.com/arborx/arborx/compare/v1.3...v1.4)
+
+**New features:**
+- Add HDBSCAN (EXPERIMENTAL) [\#777](https://github.com/arborx/ArborX/pull/777), [\#783](https://github.com/arborx/ArborX/pull/783), [\#845](https://github.com/arborx/ArborX/pull/845)
+- Add specialized "half" tree traversal for spatial predicates [\#808](https://github.com/arborx/ArborX/pull/808)
```

Comment:
not user facing so not sure it should be a "feature"

Reply:
Fair point. My take is that maybe it should be user facing at some point, given that it is really well suited for pair interactions. But you are probably right, should remove it now.